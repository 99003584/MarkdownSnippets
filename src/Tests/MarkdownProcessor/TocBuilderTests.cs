using System.Collections.Generic;
using ApprovalTests;
using Xunit;
using Xunit.Abstractions;

public class TocBuilderTests :
    TestBase
{
    [Fact]
    public void Single()
    {
        var lines = new List<Line>
        {
            new Line("## Heading", "", 0)
        };

        Approvals.Verify(TocBuilder.BuildToc(lines));
    }

    [Fact]
    public void WithSpaces()
    {
        var lines = new List<Line>
        {
            new Line("##  A B ", "", 0)
        };

        Approvals.Verify(TocBuilder.BuildToc(lines));
    }

    [Fact]
    public void Duplicates()
    {
        var lines = new List<Line>
        {
            new Line("## A", "", 0),
            new Line("## A", "", 0),
            new Line("## a", "", 0)
        };

        Approvals.Verify(TocBuilder.BuildToc(lines));
    }

    public TocBuilderTests(ITestOutputHelper output) :
        base(output)
    {
    }
}