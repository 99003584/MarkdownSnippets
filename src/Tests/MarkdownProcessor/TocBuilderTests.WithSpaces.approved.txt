<!-- toc -->
## Contents

 * [A B](#a-b)
